a10_octavia/controller/worker/tasks/cert_tasks.py

```diff
@@ -63,143 +64,153 @@ class SSLKeyCreate(task.Task):
+            except Exception:
+                pass
 
 
 class SSLCertUpdate(task.Task):
     """Task to update an SSL certificate"""
     @axapi_client_decorator
     def execute(self, loadbalancer, listener, vthunder, action="import", certificate_type="pem"):
-        barbican_client = BarbicanACLAuth().get_barbican_client(loadbalancer.project_id)
-        cert_data = utils.get_cert_data(barbican_client, listener)
-        try:
-            if self.axapi_client.file.ssl_cert.exists(file=cert_data.cert_filename):
-                self.axapi_client.file.ssl_cert.update(file=cert_data.cert_filename,
-                                                       cert=cert_data.cert_content,
-                                                       size=len(cert_data.cert_content),
-                                                       action=action,
-                                                       certificate_type=certificate_type)
-                LOG.debug("SSL certificate updated successfully: %s", cert_data.cert_filename)
-        except Exception as e:
-            LOG.warning("Failed to update SSL Certificate: %s", str(e))
-        return cert_data
+        if listener.protocol == "TERMINATED_HTTPS":
+            barbican_client = BarbicanACLAuth().get_barbican_client(loadbalancer.project_id)
+            cert_data = utils.get_cert_data(barbican_client, listener)
+            try:
+                if self.axapi_client.file.ssl_cert.exists(file=cert_data.cert_filename):
+                    self.axapi_client.file.ssl_cert.update(file=cert_data.cert_filename,
+                                                           cert=cert_data.cert_content,
+                                                           size=len(cert_data.cert_content),
+                                                           action=action,
+                                                           certificate_type=certificate_type)
+                    LOG.debug("SSL certificate updated successfully: %s", cert_data.cert_filename)
+            except Exception as e:
+                LOG.warning("Failed to update SSL Certificate: %s", str(e))
+            return cert_data
 
 
 class SSLKeyUpdate(task.Task):
     """Task to update an SSL Key"""
     @axapi_client_decorator
     def execute(self, cert_data, vthunder, action="import"):
-        try:
-            if self.axapi_client.file.ssl_key.exists(file=cert_data.key_filename):
-                self.axapi_client.file.ssl_key.update(file=cert_data.key_filename,
-                                                      cert=cert_data.key_content,
-                                                      size=len(cert_data.key_content),
-                                                      action=action)
-                LOG.debug("SSL Key updated successfully: %s", cert_data.key_filename)
-        except Exception as e:
-            LOG.warning("Failed to update SSL Key: %s", str(e))
+        if cert_data is not None:
```

Comment:
This shouldn't be needed. We've designed the flows so that cert_data will never be none if this task is executed.